### DISCUSSION TITLE
Integration with postgit

### DISCUSSION THREAD

**Discussion by FelixZY** [theory/sqitch#820] [Ideas]
Over the past few days, I've been looking for a more declarative way of managing a plain SQL project. In a perfect world, I'd prefer to define the desired state rather than how to get there. However, there seems to be few ways to do this while writing plain SQL at the moment. I've found that sqitch will probably get me closest.

In my research, I came across a [blog post](https://commit.dev/2022/12/20/declarative-version-controlled-database-migrations-my-hackathon-onboarding-project/) where the author, @matthieu-foucault , was able to get close while remaining vendor-agnostic. The product of his efforts, a proof of concept called [postgit](https://github.com/matthieu-foucault/postgit/), is available on GitHub but [seemingly unmaintained](https://github.com/matthieu-foucault/postgit/issues/22#issuecomment-2007494398).

Even so, I think his efforts were interesting and perhaps somewhat in alignment with [sqitch's goals](https://speakerdeck.com/theory/agile-database-development-3ed?slide=228) as well. Perhaps there are even ideas that could be reused by sqitch directly, such as support for automated migration generation using a diff tool?

**Comment by FelixZY** [comment]
(I guess that the implementation really is vendor specific, targeting PostgreSQL, but from a cursory view, it should be relatively simple to abstract the implementation away; all that's needed is a diff provider for the specific SQL flavour)

**Reply by theory** [reply]
Seems nicely complementary, yes, to ease the burden of writing change scripts by generating them via diffs. But yeah, since it's Postgres-specific, I'd consider it a complementary tool rather than something to be integrated. More [Unix-y in philosophy](https://en.wikipedia.org/wiki/Unix_philosophy), too :-)
